python-frontend/src/main/java/org/sonar/python/cfg/ControlFlowGraphBuilder.java

```diff
@@ -67,13 +70,41 @@ public ControlFlowGraphBuilder(@Nullable StatementList statementList) {
     exitTargets.push(end);
     if (statementList != null) {
       start = build(statementList.statements(), createSimpleBlock(end));
+      addParametersToStartBlock(statementList);
     } else {
       start = end;
     }
     removeEmptyBlocks();
     computePredecessors();
   }
 
+  private void addParametersToStartBlock(StatementList statementList) {
+    if (statementList.parent().is(Tree.Kind.FUNCDEF)) {
+      ParameterList parameterList = ((FunctionDef) statementList.parent()).parameters();
+      if (parameterList != null) {
+        PythonCfgSimpleBlock parametersBlock = createSimpleBlock(start);
```

Comment:
We have a duplication about the logic PARAMETER/TUPLE_PARAMETER. We could simplify with:
```java
        PythonCfgSimpleBlock parametersBlock = createSimpleBlock(start);
        addParameters(parameterList.all(), parametersBlock);
        start = parametersBlock;
```
...
```java
  private static void addParameters(List<AnyParameter> parameters, PythonCfgSimpleBlock parametersBlock) {
    for (AnyParameter parameter : parameters) {
      if (parameter.is(Tree.Kind.TUPLE_PARAMETER)) {
        addParameters(((TupleParameter) parameter).parameters(), parametersBlock);
      } else {
        parametersBlock.addElement(parameter);
      }
    }
  }
```